server/src/main/scala/org/http4s/server/middleware/ChunkAggregator.scala

```diff
@@ -2,19 +2,32 @@ package org.http4s
 package server
 package middleware
 
+import cats.data.NonEmptyList
 import fs2._
 import fs2.interop.cats._
 import org.http4s.headers._
 
 object ChunkAggregator {
-  def apply(service: HttpService): HttpService = service.flatMapF[MaybeResponse] {
-    case response: Response =>
-      response.body.runLog.map { bytes =>
-        if (bytes.nonEmpty) {
-          val h = response.headers.put(`Content-Length`(bytes.length.toLong))
-          response.copy(body = Stream.emits(bytes), headers = h)
-        } else response
+  def apply(service: HttpService): HttpService =
+    service.flatMapF[MaybeResponse] {
+      case Pass => Task.now(Pass)
+      case response: Response =>
+        response.body.runLog.flatMap { fullBody =>
+          if (fullBody.nonEmpty)
+            response.withBody(Chunk.seq(fullBody)).map(removeChunkedTransferEncoding)
+          else
+            Task.now(response)
+        }
+    }
+
+  private val removeChunkedTransferEncoding: Response => Response = {
+    _.transformHeaders { headers =>
+      headers.flatMap {
+        case e: `Transfer-Encoding` =>
+          NonEmptyList.fromList(e.values.filterNot(_ == TransferCoding.chunked)).map(`Transfer-Encoding`.apply).toList
```

Comment:
Took me a minute to understand this, but then I saw the property test and was pretty sure you were right, then came back and understood this.

I love property tests!